src/dataflow/src/source/s3/metrics.rs

```diff
@@ -0,0 +1,97 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use lazy_static::lazy_static;
+use prometheus::core::AtomicU64;
+use prometheus::{register_uint_counter_vec, DeleteOnDropCounter, UIntCounterVec};
+
+pub(super) struct BucketMetrics {
+    objects_downloaded: DeleteOnDropCounter<'static, AtomicU64>,
+    bytes_downloaded: DeleteOnDropCounter<'static, AtomicU64>,
+    lines_ingested: DeleteOnDropCounter<'static, AtomicU64>,
+}
+
+impl BucketMetrics {
+    pub(super) fn new(source_id: &str, bucket_id: &str) -> BucketMetrics {
+        const LABELS: &[&str] = &["bucket_id", "source_id"];
+        lazy_static! {
+            static ref OBJECTS_DOWNLOADED: UIntCounterVec = register_uint_counter_vec!(
+                "mz_s3_objects_downloaded",
+                "The number of s3 objects that we have downloaded.",
+                LABELS
+            )
+            .unwrap();
+            static ref BYTES_DOWNLOADED: UIntCounterVec = register_uint_counter_vec!(
+                "mz_s3_bytes_downloaded",
+                "The total count of bytes downloaded for this source",
+                LABELS
+            )
+            .unwrap();
+            static ref LINES_INGESTED: UIntCounterVec = register_uint_counter_vec!(
+                "mz_s3_lines_ingested",
```

Comment:
Is it lines or messages? Can messages be more than one line long?

Reply:
it is lines, currently all text sources are line-delimited. We treat lines as messages, though, so perhaps messages is the more future-proof term.

Reply:
Sounds good -- thanks for the explanation. I was confused because both `lines` and `messages` were used.